src/editpage/components/assistance/SettingsModal.vue

```diff
@@ -0,0 +1,232 @@
+            </th>
+            <th>
+              <div class="dropdown">
+                <button class="dropbtn settings-modal-button">Select</button>
+                <div class="dropdown-content">
+                  <a v-for="style in styles" :key="style"
+                      @click="changeTheme(style)">
+                      {{style}}</a>
+                </div>
+              </div>
+            </th>
+          </tr>
+        </table>
+
+      </div>
+    </div>
+</template>
+
+
+<script>
+import Vue from 'vue';
+
+export default {
+  name: 'SettingsModal',
+  components: {},
+  props: {},
+  data: function() {
+    return {
+      eventBus: new Vue(),
+      zoomMin: 0.3,
+      zoomMax: 3,
+      zoomSliderValue: 0,
+      styles: ['light', 'dark', 'light_roboto', 'dark_roboto'],
+    };
+  },
+  methods: {
+    forwardEvent: function(event) {
+      console.log(event);
+      this.eventBus.$emit(event);
+      console.log('forwarded done');
+    },
+    closeSettingsModal: function() {
+      document.getElementById('settingsModal').style.display = 'none';
+    },
+
+    openSettingsModal: function() {
+      document.getElementById('settingsModal').style.display = 'block';
+      this.updateConfigurationString();
+      // const release = await store.state.lock.acquire();
+
+      // release();
+    },
+
+    zoomIn: function() {
+      this.zoomChange(1.1);
+    },
+
+    zoomOut: function() {
+      this.zoomChange(1/1.1);
+    },
+
+    zoomSliderChanged: function() {
+      const val = parseFloat(this.zoomSliderValue);
+      this.$store.commit('setZoom', val);
+    },
+
+    zoomChange: function(factor) {
+      const newZoom = this.$store.state.settings.zoom * factor;
+
+      const boundedZoom = Math.max(0.3, Math.min(newZoom, 3.0));
+      this.$store.commit('setZoom', boundedZoom);
+      this.updateConfigurationString();
+    },
+
+    changeTheme: function(theme) {
+      if (this.styles.includes(theme)) {
+        this.$store.commit('setTheme', theme);
+      }
+    },
+
+    updateConfigurationString: function() {
+      const libs = this.$store.state.libraries;
+      const settings = this.$store.state.settings;
+      document.getElementById('settingsOverview').innerHTML =
+          'Sertop path: ' + libs.sertopPath + '<br /> Serapi Version: '
+          + libs.serapiVersion + '<br /> Library Version: '
+          + libs.libraryVersion + '<br /> Zoom: ' + settings.zoom
+          + '<br /> Theme: ' + settings.theme;
+    },
+  },
+  mounted: function() {
+    this.zoomSliderValue = this.$store.state.settings.zoom;
+    this.eventBus.$on('zoomIn', this.zoomIn);
+    this.eventBus.$on('zoomOut', this.zoomOut);
+    this.eventBus.$on('openSettingsModal', this.openSettingsModal);
+  },
+};
+</script>
+
+<style lang="scss">
+  @import '../../../assets/sass/pages/edit';
+
+  /* The Modal (background) */
+  #settingsModal {
+    background-color: rgba(255,255,255,0.5); /* Partly opaque background */
```

Comment:
Maybe a slightly darker color would make it more clear that the content behind is blocked.
Also maybe the modal could close if you click on the background.